tools/lsp-wake/lsp.cpp

```diff
@@ -538,40 +516,105 @@ class LSPServer {
         // The file was deleted => clear any stale diagnostics
         std::vector<Diagnostic> emptyDiagnostics;
         JAST fileDiagnosticsJSON = JSONConverter::fileDiagnosticsToJSON(filePath, emptyDiagnostics);
-        sendMessage(fileDiagnosticsJSON);
+        methodResult.diagnostics.children.emplace_back("", fileDiagnosticsJSON);
       }
     }
 
     if (!jfiles.children.empty()) {
       needsUpdate = true;
-      refresh("files-created-or-deleted");
+      refresh("files-created-or-deleted", methodResult);
     }
+    return methodResult;
   }
 
-  void shutdown(const JAST &receivedMessage) {
+  MethodResult shutdown(const JAST &receivedMessage) {
+    MethodResult methodResult;
     JAST message = JSONConverter::createResponseMessage(receivedMessage);
     message.add("result", JSON_NULLVAL);
     isShutDown = true;
-    sendMessage(message);
+    methodResult.response = message;
+    return methodResult;
   }
 
-  void serverExit(const JAST &_) { exit(isShutDown ? 0 : 1); }
+  MethodResult serverExit(const JAST &_) { exit(isShutDown ? 0 : 1); }
 };
 
-int main(int argc, const char **argv) {
-  std::string stdLib;
-  if (argc >= 2) {
-    stdLib = make_canonical(argv[1]);
-  } else {
-    stdLib = make_canonical(find_execpath() + "/../../share/wake/lib");
+LSPServer *lspServer = nullptr;
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten/emscripten.h>
+
+void instantiateServerInternal(const std::string &stdLib) {
+  // clang-format off
+  int isNode = EM_ASM_INT({
+    if (ENVIRONMENT_IS_NODE) {
+      return 1;
+    } else {
+      return 0;
+    }
```

Comment:
does `return ENVIRONMENT_IS_NODE` not work?

Reply:
Done